images/virtualization-artifact/pkg/common/network/network.go

```diff
@@ -47,21 +48,56 @@ type InterfaceStatus struct {
 
 type InterfaceSpecList []InterfaceSpec
 
-func CreateNetworkSpec(vmSpec virtv2.VirtualMachineSpec) InterfaceSpecList {
-	var networksSpec InterfaceSpecList
-	for id, network := range vmSpec.Networks {
-		if network.Type == virtv2.NetworksTypeMain {
+func CreateNetworkSpec(vm *virtv2.VirtualMachine, vmmacs []*virtv2.VirtualMachineMACAddress) InterfaceSpecList {
+	var (
+		all     []string
+		status  []struct{ Name, MAC string }
+		taken   = make(map[string]bool)
+		free    []string
+		res     InterfaceSpecList
+		freeIdx int
+	)
+
+	for _, v := range vmmacs {
+		if mac := v.Status.Address; mac != "" {
+			all = append(all, mac)
+		}
+	}
+	for _, n := range vm.Status.Networks {
+		status = append(status, struct{ Name, MAC string }{n.Name, n.MAC})
+		taken[n.MAC] = true
+	}
```

Comment:
Filter out main network?
```suggestion
	for _, n := range vm.Status.Networks {
		if n.Type == virtv2.NetworksTypeMain {
			continue
		}
		status = append(status, struct{ Name, MAC string }{n.Name, n.MAC})
		taken[n.MAC] = true
	}
```